src/components/LayoutTree.tsx

```diff
@@ -428,12 +532,24 @@ const TreeNode = ({
   autofocusDepth: number
 } & Pick<CSSTransitionProps, 'in'>) => {
   const [y, setY] = useState(_y)
+  const [x, setX] = useState(_x)
   const fadeThoughtRef = useRef<HTMLDivElement>(null)
+
+  // true if the last action is newThought
   const isLastActionNewThought = useSelector(state => {
     const lastPatches = state.undoPatches[state.undoPatches.length - 1]
     return lastPatches?.some(patch => patch.actions[0] === 'newThought')
   })
 
+  // true if the last action is swapParent or if the last action is not moveThought
+  const isLastActionSwapParent = useSelector(state => {
+    const lastPatches = state.undoPatches[state.undoPatches.length - 1]
+    return (
+      lastPatches?.some(patch => patch.actions[0] === 'swapParent') ||
+      lastPatches?.every(patch => patch.actions[0] !== 'moveThought')
```

Comment:
I'm not really sure what the intention is here, but these conditions are not orthogonal. If the last action is `swapParent`, then `!== 'moveThought'` is necessarily true.

Reply:
Good catch. This does not make sense. I will fix it.

Reply:
This has been reverted. I've fixed the unexpected animation after swapping parent by checking if there is any `swapParent` action from the last patches.